test/test_envmod.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from unittest.mock import patch
+
+from payu.envmod import check_modulefile
+
+
+@patch('payu.envmod.run_module_cmd')
+def test_check_modulefile_unique(mock_module_cmd):
+    # Mock module avail command
+    mock_module_cmd.return_value.stderr = """/path/to/modulefiles:
+test-module/1.0.0
+"""
+    # Test runs without an error
+    check_modulefile('test-module/1.0.0')
+
+
+@patch('payu.envmod.run_module_cmd')
+def test_check_modulefile_without_version(mock_module_cmd):
+    # Mock module avail command
+    mock_module_cmd.return_value.stderr = """/path/to/modulefiles:
+test-module/1.0.0
+test-module/2.0.0
+"""
+
+    # Expect an error raised
+    with pytest.raises(ValueError) as exc_info:
+        check_modulefile('test-module')
+        exc_info.value.startswith(
+            "There are multiple modules available for test-module"
+        )
+
+
+@patch('payu.envmod.run_module_cmd')
+def test_check_modulefile_exact_match(mock_module_cmd):
+    # Mock module avail command
+    mock_module_cmd.return_value.stderr = """/path/to/modulefiles:
+test-module/1.0.0
+test-module/1.0.0-debug
```

Comment:
We're not testing that this runs without error if the version is specified. The previous test uses different input.

Just saying logically we haven't tested the negation of this.

Reply:
Sorry I am a bit confused on what you mean. Do you mean testing there's an error when no version is specified, when there is following output?
```
test-module/1.0.0
test-module/1.0.0-debug
```
If so, I've added to the previous test.

Reply:
Sorry for not being clear. Yes that is what I was saying. There was a previous test but it wasn't exactly the same as this, so logically it wasn't clear that the test showing it ran without error did so because the version was specified, as it didn't have a matching fail state when the version wasn't specified.

Does that make sense?

Reply:
It was a bit nit-picky.

Reply:
Yes that makes sense thanks. It was good to add a test to make the expected behaviour more clear anyway